### PR TITLE
hyprland: 0.41.2 -> 0.42.0; aquamarine: 0.2.0 -> 0.3.1

### PR DESCRIPTION
**Pull request by fufexan**
## Description of changes

[Hyprland changelog](https://github.com/hyprwm/Hyprland/releases/tag/v0.42.0)
[Aquamarine changelog](github.com/hyprwm/aquamarine/releases/v0.3.0)

TODO:
- [x] update plugins

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by donovanglover** [comment]
FWIW [some hyprland-rs stuff](https://github.com/search?q=Client%3A%3Aget_active%28%29+language%3Arust+hyprland&type=code) is going to break from https://github.com/hyprwm/Hyprland/pull/7104

Fixed in https://github.com/hyprland-community/hyprland-rs/pull/258 but downstream projects will have to update to it.

**Comment by fufexan** [comment]
We also have to wait for plugins to be updated.

File: pkgs/by-name/aq/aquamarine/update.sh
Comment (JohnRTitor) [inline pkgs/by-name/aq/aquamarine/update.sh:1]: Is `nix-update-script` not working for this?

Comment (fufexan) [reply pkgs/by-name/aq/aquamarine/update.sh:1]: I'm not sure myself. I started working on this PR in a git worktree, and it seems fd has trouble finding hyprland there. But without worktrees it works just fine.

Comment (JohnRTitor) [reply pkgs/by-name/aq/aquamarine/update.sh:1]: CC @Mic92

**Comment by donovanglover** [comment]
Plugin status:

- hyprscroller: Creator is working on it https://github.com/dawsers/hyprscroller/pull/38#issuecomment-2277521385
- hycov: Missing fix for the fullscreen change https://github.com/kilzm/hycov
- hypr-dynamic-cursors: [Works](https://github.com/NixOS/nixpkgs/commit/dee8e256d738664e7cb602a0c78b7a5845ec60ed) after update
- hy3: [Works](https://github.com/NixOS/nixpkgs/commit/fe52a50292ee5a12130148bcdb87e5eae3da36ec) after update
- hyprspace: [Works](https://github.com/NixOS/nixpkgs/commit/0a9614120bcc9badef84d25477f71652f0d5c093) with https://github.com/clague/Hyprspace

**Comment by JohnRTitor** [comment]
Do we really need to hold back Hyprland update because plugins could not catch up? Hyprland plugins depend on `hyprland`, not the other way around.

File: pkgs/by-name/hy/hyprland/package.nix
Comment (JohnRTitor) [inline pkgs/by-name/hy/hyprland/package.nix:1]: Let's change the `meta.platforms` to include freebsd as well.

**Comment by Mic92** [comment]
> Do we really need to hold back Hyprland update because plugins could not catch up? Hyprland plugins depend on `hyprland`, not the other way around.

If people feel strong about those, we can have multiple hyprland versions eventually.

**Comment by Kraust** [comment]
> > Do we really need to hold back Hyprland update because plugins could not catch up? Hyprland plugins depend on `hyprland`, not the other way around.
> 
> If people feel strong about those, we can have multiple hyprland versions eventually.

Are there at least steps to override the current version of hyprland like you can do with other packages? I don't use any of these plugins and would like the latest hyprland.

**Comment by donovanglover** [comment]
Let's update Hyprland independently of plugins. I can make a separate PR to update the plugins that work and mark the other ones as broken.

File: pkgs/by-name/hy/hyprland/package.nix
Comment (fufexan) [reply pkgs/by-name/hy/hyprland/package.nix:1]: Sure, done.

**Review by JohnRTitor — COMMENTED** [review]
Where's hy3?

File: pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix
Comment (JohnRTitor) [inline pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix:1]: These should inherit the Hyprland `meta.platforms`

File: pkgs/by-name/hy/hyprland/package.nix [hunk 1]
```diff
@@ -191,6 +196,9 @@ stdenv.mkDerivation (finalAttrs: {
       wozeparrot
     ];
     mainProgram = "Hyprland";
-    platforms = lib.platforms.linux;
+    platforms = lib.concatLists [
+      lib.platforms.linux
+      lib.platforms.freebsd
+    ];
```
Comment (JohnRTitor) [inline pkgs/by-name/hy/hyprland/package.nix:202]: `lib.platforms.linux ++ lib.platforms.freebsd` would be cleaner

Comment (fufexan) [reply pkgs/by-name/hy/hyprland/package.nix]: Done

File: pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix
Comment (fufexan) [reply pkgs/applications/window-managers/hyprwm/hyprland-plugins/hyprland-plugins.nix:1]: Done.

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 333074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>hyprlandPlugins.hycov</li>
    <li>hyprlandPlugins.hypr-dynamic-cursors</li>
    <li>hyprlandPlugins.hyprscroller</li>
    <li>hyprlandPlugins.hyprspace</li>
  </ul>
</details>
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>aquamarine</li>
    <li>aquamarine.dev</li>
    <li>grimblast</li>
    <li>hdrop</li>
    <li>hyprland</li>
    <li>hyprland.dev</li>
    <li>hyprland.man</li>
    <li>hyprlandPlugins.borders-plus-plus</li>
    <li>hyprlandPlugins.csgo-vulkan-fix</li>
    <li>hyprlandPlugins.hy3</li>
    <li>hyprlandPlugins.hyprbars</li>
    <li>hyprlandPlugins.hyprexpo</li>
    <li>hyprlandPlugins.hyprfocus</li>
    <li>hyprlandPlugins.hyprgrass</li>
    <li>hyprlandPlugins.hyprtrails</li>
    <li>hyprlandPlugins.hyprwinwrap</li>
    <li>hyprshade</li>
    <li>hyprshade.dist</li>
    <li>hyprshot</li>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>waybar</li>
    <li>xdg-desktop-portal-hyprland</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
@NixOS/nixpkgs-merge-bot merge

I wonder if the merge queue is implemented

**Review by JohnRTitor — APPROVED** [review]
LGTM, let's merge this after eval passes

**Comment by eqfae** [comment]
图片
When will it sync to unstable? I found out that even if it is merged, it has not changed.

**Comment by Mic92** [comment]
@eqfae https://nixpk.gs/pr-tracker.html?pr=333074

**Pull request merged**